Sources/_StringProcessing/Regex/AnyRegexOutput.swift

```diff
@@ -145,6 +145,10 @@ extension Regex where Output == AnyRegexOutput {
   public init(_ pattern: String) throws {
     self.init(ast: try parse(pattern, .semantic, .traditional))
   }
+  
+  internal init(_ pattern: String, syntax: SyntaxOptions) throws {
+    self.init(ast: try parse(pattern, .semantic, syntax))
```

Comment:
It's because of the ordering of the changes, the parser recovery PR was written after this change, but cherry-picked before it. I resolved the conflict in my cherry-pick, as I wasn't sure whether this was going to be cherry-picked or not